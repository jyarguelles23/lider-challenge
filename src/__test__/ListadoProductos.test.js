import ListaProducto from "../components/ListadoProductos";
import React from "react";
import renderer, { ReactTestRenderer } from "react-test-renderer";
import Imagen from "../components/ImagenProducto";

let tree = ReactTestRenderer;

const productos = [
  {
    id: "5f63bf9c55b47b07c4fb83b8",
    brand: "ooy eqrceli",
    description: "rlñlw brhrka",
    image: "https://lider.cl/catalogo/images/whiteLineIcon.svg",
    price: 498724,
    oldPrice: 0,
  },
  {
    id: "5f63bf9c55b47b07c4fb83ba",
    brand: "dsaasd",
    description: "zlrwax bñyrh",
    image: "https://lider.cl/catalogo/images/babyIcon.svg",
    price: 130173,
    oldPrice: 0,
  },
  {
    id: "5f63bf9c55b47b07c4fb83bc",
    brand: "weñxoab",
    description: "hqhoy qacirk",
    image: "https://lider.cl/catalogo/images/homeIcon.svg",
    price: 171740,
    oldPrice: 0,
  },
];

describe("<ListaProducto/>", () => {
  beforeEach(() => {
    tree = renderer.create(<ListaProducto productos={productos} />);
  });

  it("should render without crashing", () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render 3 'ImagenProducto'", () => {
    const images = tree.root.findAllByType(Imagen);
    expect(images.length).toBe(3);
  });  
   
});
