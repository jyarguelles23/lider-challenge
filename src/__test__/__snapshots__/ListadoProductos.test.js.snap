// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListaProducto/> should render without crashing 1`] = `
<div
  className="col-12 p-5 row"
  data-testid="lista-productos-component"
>
  <div
    className="col-12 col-sm-6 col-md-4 col-lg-3 mb-4"
  >
    <div
      className="card"
    >
      <img
        alt="ooy eqrceli"
        className="card-img-top"
        src="https://lider.cl/catalogo/images/whiteLineIcon.svg"
      />
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
          <strong>
            ooy eqrceli
          </strong>
           
          rl単lw brhrka
           
        </p>
        <div
          className="product-price"
        >
          $ 
          498.724
        </div>
        <div
          className="product-old-price"
        />
      </div>
    </div>
  </div>
  <div
    className="col-12 col-sm-6 col-md-4 col-lg-3 mb-4"
  >
    <div
      className="card"
    >
      <img
        alt="dsaasd"
        className="card-img-top"
        src="https://lider.cl/catalogo/images/babyIcon.svg"
      />
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
          <strong>
            dsaasd
          </strong>
           
          zlrwax b単yrh
           
        </p>
        <div
          className="product-price"
        >
          $ 
          130.173
        </div>
        <div
          className="product-old-price"
        />
      </div>
    </div>
  </div>
  <div
    className="col-12 col-sm-6 col-md-4 col-lg-3 mb-4"
  >
    <div
      className="card"
    >
      <img
        alt="we単xoab"
        className="card-img-top"
        src="https://lider.cl/catalogo/images/homeIcon.svg"
      />
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
          <strong>
            we単xoab
          </strong>
           
          hqhoy qacirk
           
        </p>
        <div
          className="product-price"
        >
          $ 
          171.740
        </div>
        <div
          className="product-old-price"
        />
      </div>
    </div>
  </div>
</div>
`;
